com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/DefaultExportFeatureExtensionService.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.xtext.resource;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.naming.QualifiedName;
+import org.eclipse.xtext.resource.IEObjectDescription;
+import org.eclipse.xtext.util.IAcceptor;
+
+import com.avaloq.tools.ddk.xtext.extension.ILanguageExtensionsService;
+import com.avaloq.tools.ddk.xtext.extension.ILanguageExtensions;
+import com.google.common.collect.Lists;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Singleton;
+
+
+/**
+ * Default implementation relying on {@link ILanguageExtensions}.
+ */
+@Singleton

```

Comment:
i'd say whenever the Singleton annotation is used, the class should be thread-safe. If I understand right, this isn't that important in this context, but pls be aware of this

Reply:
Initialisation is now thread safe as it is done by Guice.